util/process_perf_data.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import ConfigParser
+import gzip
+import json
+import math
+import os
+import socket
+import sys
+import time
+import uuid
+
+from thclient import (TreeherderClient, TreeherderJob,
+                      TreeherderJobCollection, TreeherderArtifactCollection)
+
+# A description of each checkpoint and the root path to it.
+CHECKPOINTS = [
+    { 'name': "Fresh start", 'path': "Iteration 1/Start" },
+    { 'name': "Fresh start [+30s]", 'path': "Iteration 1/StartSettled" },
+    { 'name': "After tabs open", 'path': "Iteration 5/TabsOpen" },
+    { 'name': "After tabs open [+30s]", 'path': "Iteration 5/TabsOpenSettled" },
+    { 'name': "After tabs open [+30s, forced GC]", 'path': "Iteration 5/TabsOpenForceGC" },
+    { 'name': "Tabs closed", 'path': "Iteration 5/TabsClosed" },
+    { 'name': "Tabs closed [+30s]", 'path': "Iteration 5/TabsClosedSettled" },
+    { 'name': "Tabs closed [+30s, forced GC]", 'path': "Iteration 5/TabsClosedForceGC" }
+]
+
+# A description of each perfherder suite and the path to its values.
+PERF_SUITES = [
+    { 'name': "Resident Memory", 'node': "resident" },
+    { 'name': "Explicit Memory", 'node': "explicit" },
+    { 'name': "Heap Unclassified", 'node': "explicit/heap-unclassified" },
+    { 'name': "JS", 'node': "js-main-runtime" },
+    { 'name': "Images", 'node': "explicit/images" }
+]
+
+def get_node_value(datapoint, nodes):
+    """
+    Retrieves a nested value from the nodes dictionary.
```

Comment:
I would like to see an example of the datastructure here that we are parsing- I can infer it from the code, but being explicit or linking to an example in a readme would be really cool!